packages/keystone/src/fields/types/float/views/index.tsx

```diff
@@ -43,17 +161,40 @@ export const CardValue: CardValueComponent = ({ item, field }) => {
   );
 };
 
-export const controller = (config: FieldControllerConfig): FieldController<string, string> => {
+export const controller = (
+  config: FieldControllerConfig<{ validation: Validation; defaultValue: number | null }>
+): FieldController<Value, string> & {
+  validation: Validation;
+} => {
   return {
     path: config.path,
     label: config.label,
     graphqlSelection: config.path,
-    defaultValue: '',
-    deserialize: data => {
-      const value = data[config.path];
-      return typeof value === 'number' ? value + '' : '';
+    validation: config.fieldMeta.validation,
+    defaultValue: {
+      kind: 'create',
+      value: config.fieldMeta.defaultValue,
+    },
+    deserialize: data => ({
+      kind: 'update',
+      initial: data[config.path],
+      value: data[config.path],
+    }),
+    serialize: value => {
+      let vale = value.value;
+      let val;
+
+      if (vale === '' || vale === null) {
+        val = null;
+      } else if (typeof vale === 'string') {
+        val = parseFloat(vale);
```

Comment:
I apologize in advance for the comment from the sidelines: this has an issue in some locales. In Germany e.g. `,` is used as a decimal seperator, which doesn't work well with `parseFloat`.

I've recently written a small function which basically deals with this case: https://gist.github.com/gu-stav/c8ab96802d2826939e3996ba351e1f65

After sharing that within my company, someone mentioned that the Adobe Spectrum system contains already a package for that: https://www.npmjs.com/package/@internationalized/number

I'd be really happy, if this logic could become part of the field, since it's often a source of confusion. Especially if you are moving into the direction of a translated admin ui, translation and localization should be thought about together :)

Reply:
Thanks for this! This is something we've been thinking about but haven't had a good solution for. I'll look into your snippet/the library tomorrow morning (sydney time) and see if we can just plug it in to help us out.